File: 백준/Silver/1010. 다리 놓기/다리 놓기.py
import sys

test = int(sys.stdin.readline())

for i in range(test):
    N, M = map(int, sys.stdin.readline().split(' '))
    dp = [[0 for j in range(M + 1)] for i in range(M + 1)]

    for i in range(1, M + 1):
        dp[1][i] = i

        for j in range(2, i + 1):
            dp[j][i] = dp[j - 1][i - 1] + dp[j][i - 1]

    print(dp[N][M])

File: 백준/Silver/1010. 다리 놓기/다리 놓기.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int test = Integer.parseInt(br.readLine());

        for (int i = 0; i < test; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int N = Integer.parseInt(st.nextToken());
            int M = Integer.parseInt(st.nextToken());

            int[][] dp = new int[M + 1][M + 1];

            for (int j = 1; j <= M; j++) {
                dp[1][j] = j;

                for (int k = 2; k <= j; k++) {
                    dp[k][j] = dp[k - 1][j - 1] + dp[k][j - 1];
                }
            }

            System.out.println(dp[N][M]);
        }
    }
}
